src/GitVersionCore/Core/Abstractions/Git/IBranch.cs

```diff
@@ -0,0 +1,12 @@
+namespace GitVersion
+{
+    public interface IBranch
```

Comment:
Thoughts?

```suggestion
    public interface IBranch : IEquatable<IBranch>, IComparable<IBranch>
```

Reply:
Agree, I will change that in a subsequent PR